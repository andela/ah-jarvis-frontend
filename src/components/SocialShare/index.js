import React from 'react';
import PropTypes from 'prop-types';
import {
  FacebookShareButton,
  EmailShareButton,
  TwitterShareButton,
  FacebookIcon,
  TwitterIcon,
  EmailIcon,
} from 'react-share';

const SocialShare = ({ slug, title }) => (
  <React.Fragment>
    <div className="social__icon p-b--20">
      <TwitterShareButton
        url={`http://ah-jarvis-frontend.herokuapp.com/article/${slug}`}
        title={title}
        windowWidth={750}
        windowHeight={600}
        className="social--share"
      >
        <TwitterIcon size={32} round />
      </TwitterShareButton>
    </div>
    <div className="social__icon p-b--20">
      <FacebookShareButton
        url={`http://ah-jarvis-frontend.herokuapp.com/article/${slug}`}
        title={title}
        windowWidth={750}
        windowHeight={600}
        className="social--share"
      >
        <FacebookIcon size={32} round />
      </FacebookShareButton>
    </div>
    <div className="social__icon p-b--20">
      <EmailShareButton
        windowWidth={750}
        windowHeight={600}
        url={`http://ah-jarvis-frontend.herokuapp.com/article/${slug}`}
        title={title}
        className="social--share"
      >
        <EmailIcon size={32} round />
      </EmailShareButton>
    </div>
  </React.Fragment>
);

export default SocialShare;

SocialShare.propTypes = {
  slug: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
