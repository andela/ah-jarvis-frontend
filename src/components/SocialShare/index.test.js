import { shallow } from 'enzyme';
import React from 'react';
import SocialShare from '.';
import configStore from '../../store';

describe('<SocialShare />', () => {
  it('renders  <SocialShare /> components', () => {
    const snap = shallow(<SocialShare store={configStore()} location="" />);
    expect(snap).toMatchSnapshot();
  });
});
