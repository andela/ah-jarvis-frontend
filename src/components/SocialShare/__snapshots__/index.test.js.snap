// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialShare /> renders  <SocialShare /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialShare
    location=""
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="social__icon p-b--20"
>
          <CreatedButton
                    className="social--share"
                    title={undefined}
                    url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
                    windowHeight={600}
                    windowWidth={750}
          >
                    <Icon
                              className=""
                              iconBgStyle={Object {}}
                              logoFillColor="white"
                              round={true}
                              size={32}
                    />
          </CreatedButton>
</div>,
        <div
          className="social__icon p-b--20"
>
          <CreatedButton
                    className="social--share"
                    title={undefined}
                    url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
                    windowHeight={600}
                    windowWidth={750}
          >
                    <Icon
                              className=""
                              iconBgStyle={Object {}}
                              logoFillColor="white"
                              round={true}
                              size={32}
                    />
          </CreatedButton>
</div>,
        <div
          className="social__icon p-b--20"
>
          <CreatedButton
                    className="social--share"
                    onClick={[Function]}
                    openWindow={false}
                    title={undefined}
                    url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
                    windowHeight={600}
                    windowWidth={750}
          >
                    <Icon
                              className=""
                              iconBgStyle={Object {}}
                              logoFillColor="white"
                              round={true}
                              size={32}
                    />
          </CreatedButton>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CreatedButton
            className="social--share"
            title={undefined}
            url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
            windowHeight={600}
            windowWidth={750}
>
            <Icon
                        className=""
                        iconBgStyle={Object {}}
                        logoFillColor="white"
                        round={true}
                        size={32}
            />
</CreatedButton>,
          "className": "social__icon p-b--20",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={32}
/>,
            "className": "social--share",
            "title": undefined,
            "url": "http://ah-jarvis-frontend.herokuapp.com/article/undefined",
            "windowHeight": 600,
            "windowWidth": 750,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "iconBgStyle": Object {},
              "logoFillColor": "white",
              "round": true,
              "size": 32,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CreatedButton
            className="social--share"
            title={undefined}
            url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
            windowHeight={600}
            windowWidth={750}
>
            <Icon
                        className=""
                        iconBgStyle={Object {}}
                        logoFillColor="white"
                        round={true}
                        size={32}
            />
</CreatedButton>,
          "className": "social__icon p-b--20",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={32}
/>,
            "className": "social--share",
            "title": undefined,
            "url": "http://ah-jarvis-frontend.herokuapp.com/article/undefined",
            "windowHeight": 600,
            "windowWidth": 750,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "iconBgStyle": Object {},
              "logoFillColor": "white",
              "round": true,
              "size": 32,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CreatedButton
            className="social--share"
            onClick={[Function]}
            openWindow={false}
            title={undefined}
            url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
            windowHeight={600}
            windowWidth={750}
>
            <Icon
                        className=""
                        iconBgStyle={Object {}}
                        logoFillColor="white"
                        round={true}
                        size={32}
            />
</CreatedButton>,
          "className": "social__icon p-b--20",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={32}
/>,
            "className": "social--share",
            "onClick": [Function],
            "openWindow": false,
            "title": undefined,
            "url": "http://ah-jarvis-frontend.herokuapp.com/article/undefined",
            "windowHeight": 600,
            "windowWidth": 750,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "iconBgStyle": Object {},
              "logoFillColor": "white",
              "round": true,
              "size": 32,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="social__icon p-b--20"
>
            <CreatedButton
                        className="social--share"
                        title={undefined}
                        url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
                        windowHeight={600}
                        windowWidth={750}
            >
                        <Icon
                                    className=""
                                    iconBgStyle={Object {}}
                                    logoFillColor="white"
                                    round={true}
                                    size={32}
                        />
            </CreatedButton>
</div>,
          <div
            className="social__icon p-b--20"
>
            <CreatedButton
                        className="social--share"
                        title={undefined}
                        url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
                        windowHeight={600}
                        windowWidth={750}
            >
                        <Icon
                                    className=""
                                    iconBgStyle={Object {}}
                                    logoFillColor="white"
                                    round={true}
                                    size={32}
                        />
            </CreatedButton>
</div>,
          <div
            className="social__icon p-b--20"
>
            <CreatedButton
                        className="social--share"
                        onClick={[Function]}
                        openWindow={false}
                        title={undefined}
                        url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
                        windowHeight={600}
                        windowWidth={750}
            >
                        <Icon
                                    className=""
                                    iconBgStyle={Object {}}
                                    logoFillColor="white"
                                    round={true}
                                    size={32}
                        />
            </CreatedButton>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CreatedButton
              className="social--share"
              title={undefined}
              url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
              windowHeight={600}
              windowWidth={750}
>
              <Icon
                            className=""
                            iconBgStyle={Object {}}
                            logoFillColor="white"
                            round={true}
                            size={32}
              />
</CreatedButton>,
            "className": "social__icon p-b--20",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Icon
                className=""
                iconBgStyle={Object {}}
                logoFillColor="white"
                round={true}
                size={32}
/>,
              "className": "social--share",
              "title": undefined,
              "url": "http://ah-jarvis-frontend.herokuapp.com/article/undefined",
              "windowHeight": 600,
              "windowWidth": 750,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "iconBgStyle": Object {},
                "logoFillColor": "white",
                "round": true,
                "size": 32,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CreatedButton
              className="social--share"
              title={undefined}
              url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
              windowHeight={600}
              windowWidth={750}
>
              <Icon
                            className=""
                            iconBgStyle={Object {}}
                            logoFillColor="white"
                            round={true}
                            size={32}
              />
</CreatedButton>,
            "className": "social__icon p-b--20",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Icon
                className=""
                iconBgStyle={Object {}}
                logoFillColor="white"
                round={true}
                size={32}
/>,
              "className": "social--share",
              "title": undefined,
              "url": "http://ah-jarvis-frontend.herokuapp.com/article/undefined",
              "windowHeight": 600,
              "windowWidth": 750,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "iconBgStyle": Object {},
                "logoFillColor": "white",
                "round": true,
                "size": 32,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CreatedButton
              className="social--share"
              onClick={[Function]}
              openWindow={false}
              title={undefined}
              url="http://ah-jarvis-frontend.herokuapp.com/article/undefined"
              windowHeight={600}
              windowWidth={750}
>
              <Icon
                            className=""
                            iconBgStyle={Object {}}
                            logoFillColor="white"
                            round={true}
                            size={32}
              />
</CreatedButton>,
            "className": "social__icon p-b--20",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Icon
                className=""
                iconBgStyle={Object {}}
                logoFillColor="white"
                round={true}
                size={32}
/>,
              "className": "social--share",
              "onClick": [Function],
              "openWindow": false,
              "title": undefined,
              "url": "http://ah-jarvis-frontend.herokuapp.com/article/undefined",
              "windowHeight": 600,
              "windowWidth": 750,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "iconBgStyle": Object {},
                "logoFillColor": "white",
                "round": true,
                "size": 32,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
